nilearn/decoding/tests/test_decoder.py

```diff
@@ -0,0 +1,314 @@
+
+
+# Regression
+ridge = RidgeCV()
+svr = SVR(kernel='linear')
+# Classification
+svc = LinearSVC()
+logistic_l1 = LogisticRegression(penalty='l1')
+logistic_l2 = LogisticRegression(penalty='l2')
+ridge_classifier = RidgeClassifierCV()
+random_forest = RandomForestClassifier()
+
+regressors = {'ridge': (ridge, []),
+              'svr': (svr, 'C')}
+classifiers = {'svc': (svc, 'C'),
+               'logistic_l1': (logistic_l1, 'C'),
+               'logistic_l2': (logistic_l2, 'C'),
+               'ridge_classifier': (ridge_classifier, [])}
+# Create a test dataset
+rand = np.random.RandomState(0)
+X = rand.rand(100, 10)
+# Create different targets
+y_regression = rand.rand(100)
+y_classification = np.hstack([[-1] * 50, [1] * 50])
+y_classification_str = np.hstack([['face'] * 50, ['house'] * 50])
+y_multiclass = np.hstack([[0] * 35, [1] * 30, [2] * 35])
+
+
+def test_check_param_grid():
+    """testing several estimators, each one with its specific regularization
+    parameter
+    """
+
+    # Regression
+    for _, (regressor, param) in regressors.items():
+        param_grid = _check_param_grid(regressor, X, y_regression, None)
+        assert list(param_grid.keys()) == list(param)
+    # Classification
+    for _, (classifier, param) in classifiers.items():
+        param_grid = _check_param_grid(classifier, X, y_classification, None)
+        assert list(param_grid.keys()) == list(param)
+
+    # Using a non-linear estimator to raise the error
+    for estimator in ['log_l1', random_forest]:
+        pytest.raises(ValueError, _check_param_grid, estimator, X,
+                      y_classification, None)
+
+
+def test_check_inputs_length():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    y = 2 * (y > 0) - 1
+    X_, mask = to_niimgs(X, (2, 2, 2))
+
+    # Remove ten samples from y
+    y = y[:-10]
+
+    for model in [DecoderRegressor, Decoder]:
+        pytest.raises(ValueError, model(mask=mask,
+                                        screening_percentile=100.).fit, X_, y)
+
+
+def test_check_estimator():
+    """Check if the estimator is one of the supported estimators, and if not,
+    if it is a string, and if not, then raise the error
+    """
+
+    supported_estimators = ['svc', 'svc_l2', 'svc_l1',
+                            'logistic', 'logistic_l1', 'logistic_l2',
+                            'ridge', 'ridge_classifier',
+                            'ridge_regressor', 'svr']
+    unsupported_estimators = ['ridgo', 'svb']
+    expected_warning = ('Use a custom estimator at your own risk '
+                        'of the process not working as intended.')
+
+    with warnings.catch_warnings(record=True) as raised_warnings:
+        for estimator in supported_estimators:
+            _check_estimator(_BaseDecoder(estimator=estimator).estimator)
+    warning_messages = [str(warning.message) for warning in raised_warnings]
+    assert expected_warning not in warning_messages
```

Comment:
Use the same `with pytest.warns` style here, as I had indicted in another location.
Start the `for` loop before the `with`

Reply:
But here the purpose is to check if we use correct estimator, the expected_warning would not be raised, hence we have to catch all warning messages?

Reply:
Ah okay never mind then